### PR TITLE
cpack: build: msi: Support MSI installer with CPack WiX generator

### PR DESCRIPTION
**Pull request by cosmo0920**
Currently, Fluent Bit provides NSIS installer for Windows.
But, this installer does not have the following functionalities:

* Register and Launch Fluent Bit as Windows Service during installation
* Stop and unregister Fluent Bit as Windows Service during uninstallation

Related to #5338 

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [N/A] Example configuration file for the change
- [N/A] Debug log output from testing the change

- [N/A] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

This patch can be tested as (for Windows x64):

```cmd
% cd build
% "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvars64.bat"
% cmake -DCPACK_GENERATOR=WIX ..
% cmake --build . 
% "C:\Program Files\CMake\bin\cpack.exe" -G WIX
```

Then, `fluent-bit-*.msi` will be created on building directory.

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [N/A] Attached [local packaging test](./packaging/local-build-all.sh) output showing all targets (including any new ones) build.

**Documentation**

- [x] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

**Comment by edsiper** [comment]
@cosmo0920 is this ready to go ?

**Comment by cosmo0920** [comment]
Yes. We cem merge this into master.

**Review by niedbalski — APPROVED** [review]
LGTM, minor nitpick that can be fixed later.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1055,6 +1055,16 @@ if(CPACK_GENERATOR MATCHES "NSIS")
   endif()
 endif()
 
+# CPack: Windows System w/ WiX
+if(CPACK_GENERATOR MATCHES "WIX")
+  set(CPACK_WIX_UPGRADE_GUID cb6825fd-37e6-4596-a55d-6d490d4fe178)
```
Comment (niedbalski) [inline CMakeLists.txt:1060]: place a comment indicating from where this unique id comes from or maybe reference https://cmake.org/cmake/help/v3.3/module/CPackWIX.html

**Pull request merged**